packages/yoga/src/Snackbar/web/Snackbar.jsx

```diff
@@ -0,0 +1,197 @@
+  
+      box-shadow: ${snackbar.shadow.default};
+
+      background-color: ${snackbar.variant.color[variant]};
+
+      z-index: 100;
+    `}
+
+  ${media.md`
+    left: auto;
+    bottom: ${theme.components.snackbar.position.desktop.bottom}px;
+    right: ${theme.components.snackbar.position.desktop.right}px;
+
+    width: ${theme.components.snackbar.minWidth.desktop}px;
+  `}
+`;
+
+const Snackbar = ({
+  open,
+  autoClose,
+  variant,
+  hideIcon,
+  message,
+  actionLabel,
+  onAction,
+  onClose,
+  theme: {
+    yoga: {
+      components: { snackbar },
+    },
+  },
+  ...props
+}) => {
+  const timeoutRef = useRef();
+
+  useEffect(() => {
+    const shouldCloseOnTimer = open && autoClose && onClose;
+
+    if (shouldCloseOnTimer) {
+      timeoutRef.current = setTimeout(() => {
+        onClose();
+      }, autoClose);
+    }
+
+    return () => clearTimeout(timeoutRef.current);
+  }, [open, autoClose]);
+
+  return (
+    open && (
+      <StyledSnackbar
+        role="alert"
+        aria-label={variant}
+        variant={variant}
+        {...props}
+      >
+        {!hideIcon && (
+          <IconWrapper role="img">
+            <Icon
+              as={snackbar.variant.icon[variant]}
+              fill="secondary"
+              width="large"
+              height="large"
+            />
+          </IconWrapper>
+        )}
+
+        <Box as={Text.Small} flex={1} mr="small" numberOfLines={2}>
+          {message}
+        </Box>
+
+        <ActionsWrapper>
+          {onAction && actionLabel && (
+            <Button.Link onClick={onAction} secondary small>
+              {actionLabel}
+            </Button.Link>
+          )}
+
+          {!autoClose && onClose && (
+            <IconButtonWrapper role="button" onClick={onClose}>
+              <Icon as={Close} fill="secondary" width="large" height="large" />
+            </IconButtonWrapper>
+          )}
+        </ActionsWrapper>
+      </StyledSnackbar>
+    )
+  );
+};
+
+Snackbar.propTypes = {
+  /** Controls the snackbar visibility. */
+  open: bool,
+
+  /** A number in milliseconds to close snackbar automaticaly. The `onClose` function becomes required when passing this function. */
+  autoClose: number,
+
+  /** Label for a custom action. */
+  actionLabel: string,
+
+  /** Controls the snackbar icon visibility. */
+  hideIcon: bool,
+
+  /** The message shown when snackbar is opened. */
+  message: string.isRequired,
+
+  /** Function for the custom action. The `actionLabel` becomes required when passing this function. */
+  onAction: func,
+
+  /** Function to close the snackbar. */
+  onClose: func,
+
+  /** The style variant, it may be "success", "failure" or "info". */
+  variant: oneOf(['success', 'failure', 'info']),
+};
+
+Snackbar.defaultProps = {
+  open: false,
+  autoClose: undefined,
+  actionLabel: undefined,
+  hideIcon: false,
+  onAction: undefined,
+  onClose: undefined,
+  variant: 'info',
+};
+
+export default withTheme(Snackbar);
```

Comment:
Makes sense to use Memoization here?

Reply:
This is a nice catch! Parents may render its children using the same props many times, so `React.memo` fits well